provers/evm-prover/src/main.rs

```diff
@@ -12,55 +21,202 @@ pub mod prover {
+                .expect("CELESTIA_NODE_AUTH_TOKEN must be set"),
+        };
+        let namespace_hex =
+            std::env::var("CELESTIA_NAMESPACE").expect("CELESTIA_NAMESPACE must be set");
+        let namespace = Namespace::new_v0(&hex::decode(namespace_hex)?)?;
+        let celestia_client = CelestiaClient::new(celestia_config, namespace).await?;
+
+        let prover = BlockProver::new(celestia_client, prover_config, aggregator_config);
+
+        let custom_beneficiary = env::var("CUSTOM_BENEFICIARY").ok();
+        let opcode_tracking = env::var("OPCODE_TRACKING").is_ok();
+
+        let genesis_path = std::env::var("GENESIS_PATH").expect("GENESIS_PATH must be set");
+        let genesis_json = fs::read_to_string(&genesis_path)?;
+
+        let genesis = Genesis::Custom(genesis_json);
 
         Ok(ProverService {
-            sp1_proving_key: pk,
+            evm_rpc_url,
+            evm_client,
+            prover,
+            simapp_client,
+            indexer_url,
+            genesis,
+            custom_beneficiary,
+            opcode_tracking,
         })
     }
+
+    async fn get_latest_height(&self) -> Result<ethers::types::U64, Status> {
+        self.evm_client
+            .get_block(BlockNumber::Latest)
+            .await
+            .map_err(|e| Status::internal(format!("Failed to get latest block: {}", e)))?
+            .ok_or_else(|| Status::internal("No latest block found"))?
+            .number
+            .ok_or_else(|| Status::internal("Block has no number"))
+    }
+
+    async fn get_trusted_height(&self, client_id: &str) -> Result<u64, Status> {
+        let request = tonic::Request::new(QueryClientStateRequest {
+            client_id: client_id.to_string(),
+        });
+
+        let response = self
+            .simapp_client
+            .clone()
+            .client_state(request)
+            .await?
+            .into_inner();
+
+        let client_state_json = response
+            .client_state
+            .ok_or_else(|| Status::internal("Failed to query client state"))?;
+        let client_state = ClientState::decode(client_state_json.value.as_slice())
+            .map_err(|e| Status::internal(format!("Failed to decode client state: {}", e)))?;
+        Ok(client_state.latest_height)
+    }
 }
 
 #[tonic::async_trait]
 impl Prover for ProverService {
     async fn info(&self, _request: Request<InfoRequest>) -> Result<Response<InfoResponse>, Status> {
-        let state_transition_verifier_key = self.sp1_proving_key.vk.bytes32();
-        // Empty string membership verifier key because currently membership proofs are not supported
-        let state_membership_verifier_key = String::new();
         let response = InfoResponse {
-            state_membership_verifier_key,
-            state_transition_verifier_key,
+            state_membership_verifier_key: "".to_string(),
+            state_transition_verifier_key: "".to_string(),
```

Comment:
[question] is there a TODO for returning populated verifier keys from this endpoint?

Reply:
Follow up #248